examples/LLM_Workflows/knowledge_retrieval/arxiv_articles.py

```diff
@@ -28,143 +26,74 @@ def arxiv_search_results(
-                text=_title,
-                model_name=embedding_model_name,
-            )
-            for _, _title in title.items()
-        ]
-        for future in tqdm(
-            concurrent.futures.as_completed(futures),
-            total=len(futures),
-            desc="Generating embeddings",
-        ):
-            title, embedding = future.result()
-            embedding_list.append(embedding)
-            index_list.append(title)
-
-    return pd.Series(embedding_list, index=index_list)
-
-
-def arxiv_pdfs(
-    arxiv_search_results: List[arxiv.Result], data_dir: str, max_num_concurrent_requests: int
-) -> pd.Series:
+    response = openai.Embedding.create(input=title, model=embedding_model_name)
+    return response["data"][0]["embedding"]
+
+
+def arxiv_pdf(arxiv_search_result: arxiv.Result, data_dir: str) -> str:
     """Processes the arxiv search results and downloads the PDFs.
 
-    :param arxiv_search_results: list of arxiv.Result objects.
+    :param arxiv_search_result: search result object.
     :param data_dir: the directory to save the PDFs to.
-    :param max_num_concurrent_requests: the maximum number of concurrent requests.
-    :return: a pd.Series of the filepaths to the PDFs, indexed by title.
+    :return: The filepath to the pdf after downloading
     """
-    path_list = []
-    index_list = []
-    with concurrent.futures.ThreadPoolExecutor(max_workers=max_num_concurrent_requests) as executor:
-        futures = {
-            executor.submit(
-                result.download_pdf,
-                dirpath=data_dir,
-            ): result.title
-            for result in arxiv_search_results
-        }
-        for future in tqdm(
-            concurrent.futures.as_completed(futures.keys()),
-            total=len(futures),
-            desc="Saving PDFs",
-        ):
-            filepath = future.result()
-            path_list.append(filepath)
-            index_list.append(futures[future])
-
-    return pd.Series(path_list, index=index_list)
-
-
-def arxiv_result_df(
-    title: pd.Series,
-    summary: pd.Series,
-    article_url: pd.Series,
-    pdf_url: pd.Series,
-    arxiv_pdfs: pd.Series,
-    arxiv_result_embeddings: pd.Series,
-) -> pd.DataFrame:
-    """Creates dataframe representing the arxiv search results.
-
-    :param title:
-    :param summary:
-    :param article_url:
-    :param pdf_url:
-    :param arxiv_pdfs: the location of the PDFs
-    :param arxiv_result_embeddings:  the embeddings of the titles
-    :return: a dataframe indexed by title with columns for pdf_path and embeddings
+    if not os.path.exists(data_dir):
+        os.path.makedirs(data_dir)
+    return arxiv_search_result.download_pdf(dirpath=data_dir)
+
+
+def arxiv_processed_result(
+    title: str,
+    summary: str,
+    article_url: str,
+    pdf_url: str,
+    arxiv_pdf: str,
+    arxiv_result_embedding: list[float],
+) -> Dict[str, str]:
+    """creates dict with parameters as keys/values"""
```

Comment:
This is required right -- since the collect function can only have a single `collect` ? i.e. it exists because of Parallelizeable + Collect?

Reply:
Yeah, a caveat that I want to kill at some point